app/views/dashboard/autoyast.xml.erb

```diff
@@ -9,7 +9,12 @@
         <interpreter>shell</interpreter>
         <source><![CDATA[
         #!/bin/sh
-        echo "master: <%= @controller_node %>" > /etc/salt/minion.d/master.conf
+        MINION_ID=$(grep id /tmp/master-id | sed 's/id: //')
+        if [ -z "$MINION_ID" ]; then
+            UUID=$(uuidgen -r)
+            echo "id: $UUID" > /etc/salt/minion.d/master.conf      
+            echo "master: <%= @controller_node %>" > /etc/salt/minion.d/master.conf
```

Comment:
You need to append, not overwrite. Use `>>` instead of `>` or the previous `id` redirection will be void.